UI 7

gemansipator opened the pull request:
Не разобрался в REDUX толком. Списал чужой код и приспособил к заданию. Знаю что надо переделывать

gemansipator committed:
17

gemansipator committed:
17

gemansipator committed:
Создать store в redux (папка core), который позволяет добавлять и удалять информацию о дисциплинах. Он будет включать в себя массив со строками ([“html“, “javascript“, “python“])

Данный SubjectsStore подключается на странице с курсами и отображает дисциплины в виде списка.

Дисциплины

html
js
python
‌

Рядом со списком дисциплин создать кнопу “добавить дисциплину“, при нажатии на которую выпадает модальное окошко с одним полем ввода текста (название дисциплины). При нажатии на кнопку формы “добавить” добавлять дисциплину в redux store (SubjectStore) при помощи действия (action) и метода dispatch в компоненте.

Предусмотреть возможность удалить дисциплину (крестик или иконка корзины или кнопка…)

gemansipator committed:
.

gemansipator committed:
Создать store в redux (папка core), который позволяет добавлять и удалять информацию о дисциплинах. Он будет включать в себя массив со строками ([“html“, “javascript“, “python“])

Данный SubjectsStore подключается на странице с курсами и отображает дисциплины в виде списка.

Дисциплины

html
js
python
‌

Рядом со списком дисциплин создать кнопу “добавить дисциплину“, при нажатии на которую выпадает модальное окошко с одним полем ввода текста (название дисциплины). При нажатии на кнопку формы “добавить” добавлять дисциплину в redux store (SubjectStore) при помощи действия (action) и метода dispatch в компоненте.

Предусмотреть возможность удалить дисциплину (крестик или иконка корзины или кнопка…)

gemansipator committed:
редюсер Disciplines

gemansipator committed:
Disciplines

gemansipator committed:
Не разобрался в REDUX толком. Списал чужой код и приспособил к заданию

Lekso4ka commented:
К редаксу вопросов нет - там все прекрасно!
Есть вопросы к компоненту с дисциплиной - там нет редакса...

gemansipator committed:
Список создан, не оформлен. Модальное окно добавляет дисциплину, но не сбрасывается при нажатии на ок

gemansipator committed:
Список создан, не оформлен. Модальное окно добавляет дисциплину, но не сбрасывается при нажатии на ок

gemansipator committed:
Список создан, не оформлен. Модальное окно добавляет дисциплину, но не сбрасывается при нажатии на ок

gemansipator committed:
Восстановил страницы

gemansipator committed:
Восстановил страницы

gemansipator committed:
discipline.trim()

gemansipator committed:
const handleOk = () => {
        if (discipline.trim()){
            dispatch(addDisciplines(discipline))
            setIsModalOpen(false)
        }
    };

gemansipator committed:
сделал списком

gemansipator committed:
импортировал api

gemansipator committed:
запушиваю всё, что сделал в слиянии, в ветке своей
